src/EnergyPlus/HVACDXSystem.cc

```diff
@@ -2180,186 +2198,154 @@ namespace HVACDXSystem {
-                                                              On,
-                                                              PartLoadFrac,
-                                                              SpeedNum,
-                                                              SpeedRatio,
-                                                              QZnReq,
-                                                              QLatReq,
-                                                              OnOffAirFlowRatio);
-
-                                        TempSpeedOut = Node(InletNode).MassFlowRate * (PsyHFnTdbW(Node(OutletNode).Temp, Node(OutletNode).HumRat) -
-                                                                                       PsyHFnTdbW(Node(InletNode).Temp, Node(OutletNode).HumRat));
-                                        TempSpeedReqst = Node(InletNode).MassFlowRate * (PsyHFnTdbW(DesOutTemp, Node(OutletNode).HumRat) -
-                                                                                         PsyHFnTdbW(Node(InletNode).Temp, Node(OutletNode).HumRat));
-
-                                        if ((TempSpeedOut - TempSpeedReqst) < Acc) {
-                                            SpeedNum = I;
-                                            FullOutput = TempSpeedOut; // save in case warning message is reported
-                                            break;
+                            CycRatio = 1.0;
+                            QZnReq = 0.001; // to indicate the coil is running
+                            SimVariableSpeedCoils(CompName,
+                                                  VSCoilIndex,
+                                                  FanOpMode,
+                                                  MaxONOFFCyclesperHour,
+                                                  HPTimeConstant,
+                                                  FanDelayTime,
+                                                  On,
+                                                  PartLoadFrac,
+                                                  SpeedNum,
+                                                  SpeedRatio,
+                                                  QZnReq,
+                                                  QLatReq,
+                                                  OnOffAirFlowRatio);
+
+                            TempSpeedOut = Node(InletNode).MassFlowRate * (PsyHFnTdbW(Node(OutletNode).Temp, Node(OutletNode).HumRat) -
+                                                                           PsyHFnTdbW(Node(InletNode).Temp, Node(OutletNode).HumRat));
+                            TempSpeedReqst = Node(InletNode).MassFlowRate * (PsyHFnTdbW(DesOutTemp, Node(OutletNode).HumRat) -
+                                                                             PsyHFnTdbW(Node(InletNode).Temp, Node(OutletNode).HumRat));
+
+                            if ((TempSpeedReqst - TempSpeedOut) > Acc) {
```

Comment:
Test full load result at speed 1 (not max speed) and if it meets load, then SolveRoot on answer by calling VSCoilCyclingRatio function